libs/cardano-ledger-core/src/Cardano/Ledger/CertState.hs

```diff
@@ -246,8 +247,10 @@ data PState era = PState
   }
   deriving (Show, Eq, Generic)
 
-instance NoThunks (PState era)
-
+deriving via
+  AllowThunksIn '["psDeposits"] (PState era)
```

Comment:
Why is `psDeposits` allowed to have thunks?

Reply:
I wrote some explanation in the description of the PR: 

```
Changing utxosDeposited in UTxOState to not be lazy is causing the (nightly run) NoThunks test to fail.
Simply changing the NoThunks implementation in UTxOState to allow thunks in utxosDeposited is not enough though, it seems that we have to allow them for psDeposits in PState.
I don't understand exactly why this is the case, given that PState is part of CertState, which is defined alongside with UTxOState inside LedgerState, but I guess it is a consequence of how the rules and assertions are invoked,

This PR simply makes the NoThunks test pass again, but I'm not convinced :

Is there a better way to make the test pass (instead of allowing thunks in PState)
If we have to allow thunks in PState, do we still want to have utxosDeposited strict?

```

Any suggestion welcome !